lib/src/builder/generator.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:async';
+
+import 'package:analyzer/analyzer.dart';
+import 'package:build/build.dart';
+
+import 'package:over_react/src/builder/generation/declaration_parsing.dart';
+import 'package:over_react/src/builder/generation/impl_generation.dart';
+import 'package:source_gen/source_gen.dart';
+import 'package:source_span/source_span.dart';
+
+import './util.dart';
+
+class OverReactGenerator extends Generator {
+  String _generateForFile(AssetId inputId, String primaryInputContents, CompilationUnit resolvedUnit) {
+    final sourceFile = new SourceFile.fromString(
+        primaryInputContents, url: idToPackageUri(inputId));
+
+    ImplGenerator generator;
+    if (ParsedDeclarations.mightContainDeclarations(primaryInputContents)) {
+      final declarations = new ParsedDeclarations(resolvedUnit, sourceFile, log);
+
+      if (declarations.hasErrors) {
+        log.severe('There was an error parsing the file declarations for file: $inputId');
+        return '';
```

Comment:
The convention that I've seen with builders is to populate the generated file with a block comment containing the same debug info that is logged to the terminal. That way if a user happens to open the file in their IDE, they see what happened instead of just a blank file. I don't think we need to do that in this PR, but it would be a nice quality-of-life improvement to follow up with.

Reply:
https://jira.atl.workiva.net/browse/AF-4001